.github/workflows/build-and-deploy.yml

```diff
@@ -272,8 +308,19 @@ jobs:
           maven-central-password: ${{ secrets.MAVEN_CENTRAL_PASSWORD }}
           signing-key: ${{ secrets.SIGNING_KEY }}
           signing-password: ${{ secrets.SIGNING_PASSWORD }}
-      #Add a step to publish to AUR
-
+      - name: Generate PKGBUILD
+        run: |
+          ./gradlew generatePKGBUILD
+          echo "VERSION=$(./gradlew properties | grep ^version: | cut -d" " -f2)" >> $GITHUB_ENV
+      - name: Publish AUR package
+        uses: KSXGitHub/github-actions-deploy-aur@v2.7.0
+        with:
+          pkgname: alchemist
+          pkgbuild: build/package/PKGBUILD
+          commit_username: ${{ secrets.AUR_USERNAME }}
+          commit_email: ${{ secrets.AUR_EMAIL }}
+          ssh_private_key: ${{ secrets.AUR_SSH_PRIVATE_KEY }}
+          commit_message: Update to version ${{ env.VERSION }}
```

Comment:
this stuff should be executed by release.config.js

Reply:
Both the publishing and script generation steps? 
Can I run an action step and have access to the repository secrets in release.config.js?

Reply:
We could set environmental variables with the secrets in the release job.
```
env:
  aur_username: ${{ secrets.AUR_USERNAME }}
  aur_commit_mail: ${{ secrets.AUR_COMMIT_MAIL }}
```
Then publishCmd can run a custom script that does what github-actions-deploy-aur@v2.7.0 does using those variables.

`./publishToAUR.sh`

Environmental  variables should be accessible from `publisCmd`, shouldn't they?

Reply:
> Both the publishing and script generation steps? Can I run an action step and have access to the repository secrets in release.config.js?

the script generation and testing can happen always. The publication only when there is a release.

You cannot run an action step in release.config.js, unfortunately. We will need a workaround.

Passing secrets is not a big deal (it's being done already). There is documentation on the build-and-deploy-gradle action on how to pass custom secrets.